File: BACKEND/patent/src/main/java/com/example/patent/dto/PatentRequestDTO.java
package com.example.patent.dto;


import com.example.patent.model.Prijava;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.xml.bind.annotation.XmlRootElement;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@XmlRootElement(name="patentPrijava")
public class PatentRequestDTO {

    private String potpis;
    private TNazivPronalaskaDTO nazivPronalaska;
    private TLiceDTO podnosilacPrijave;
    private TLiceDTO pronalazac;
    private TLiceDTO punomocnik;
    private AdresaDTO adresaZaDostavljanje;
    private NacinDostavljanjaDTO nacinDostavljanja;
    private String tipPrijave;
    private TDetaljiPrijaveDTO prvobitnaPrijava;
    private PrethodnePrijaveDTO prethodnePrijave;
    private boolean dodatniPodaci;

    public PatentRequestDTO(Prijava prijava){
        this.potpis = prijava.getDetaljiPrijave().getPotpis();
        this.nazivPronalaska = new TNazivPronalaskaDTO(prijava.getZahtev().getNazivPronalaska());
        this.podnosilacPrijave = new TLiceDTO(prijava.getZahtev().getPodnosilacPrijave());
        this.pronalazac = new TLiceDTO(prijava.getZahtev().getPronalazac());
        this.punomocnik = new TLiceDTO(prijava.getZahtev().getPunomocnik());
        this.adresaZaDostavljanje = new AdresaDTO(prijava.getZahtev().getAdresaZaDostavljanje());
        this.nacinDostavljanja = new NacinDostavljanjaDTO(prijava.getZahtev().getNacinDostavljanja());
        this.tipPrijave = prijava.getZahtev().getTipPrijave();
        this.prvobitnaPrijava = new TDetaljiPrijaveDTO(prijava.getZahtev().getPrvobitnaPrijava());
        this.prethodnePrijave = new PrethodnePrijaveDTO(prijava.getZahtev().getPrethodnePrijave());
        this.dodatniPodaci = prijava.getZahtev().isDodatniPodaci();


    }

    @Override
    public String toString() {
        return "PatentRequestDTO{" +
                "potpis='" + potpis + '\'' +
                ", nazivPronalaska=" + nazivPronalaska +
                ", podnosilacPrijave=" + podnosilacPrijave +
                ", pronalazac=" + pronalazac +
                ", punomocnik=" + punomocnik +
                ", adresaZaDostavljanje=" + adresaZaDostavljanje +
                ", nacinDostavljanja=" + nacinDostavljanja +
                ", tipPrijave='" + tipPrijave + '\'' +
                ", prvobitnaPrijava=" + prvobitnaPrijava +
                ", prethodnePrijave=" + prethodnePrijave +
                ", dodatniPodaci=" + dodatniPodaci +
                '}';
    }
}

File: xmlIWebServisi/src/app/model/p1/PatentRequestDTO.ts
import { AdresaDTO } from './AdresaDTO';
import { NacinDostavljanjaDTO } from './NacinDostavljanjaDTO';
import { PrethodnePrijaveDTO } from './PrethodnePrijaveDTO';
import { TDetaljiPrijaveDTO } from './TDetaljiPrijaveDTO';
import { TLiceDTO } from './TLiceDTO';
import { TNazivPronalaskaDTO } from './TNazivPronalaskaDTO';

export class PatentRequestDTO {
  public potpis: string = '';
  public nazivPronalaska: TNazivPronalaskaDTO = new TNazivPronalaskaDTO();
  public podnosilacPrijave: TLiceDTO = new TLiceDTO();
  public pronalazac: TLiceDTO = new TLiceDTO();
  public punomocnik: TLiceDTO = new TLiceDTO();
  public adresaZaDostavljanje: AdresaDTO = new AdresaDTO();
  public nacinDostavljanja: NacinDostavljanjaDTO = new NacinDostavljanjaDTO();
  public tipPrijave: string = '';
  public prvobitnaPrijava: TDetaljiPrijaveDTO = new TDetaljiPrijaveDTO();
  public prethodnePrijave: PrethodnePrijaveDTO = new PrethodnePrijaveDTO();
  public dodatniPodaci: boolean = false;
}
